include/librealsense2/dds/dds-device-broadcaster.cpp

```diff
@@ -224,11 +218,14 @@ bool dds_device_broadcaster::create_device_writer( const std::string & device_ke
 
     std::shared_ptr< dds_client_listener > writer_listener = std::make_shared< dds_client_listener >( this );
 
-    _device_handle_by_sn[device_key] = { rs2_device,
-                                         _publisher->create_datawriter( _topic, wqos, writer_listener.get() ),
-                                         writer_listener };
+    auto it_inserted = _device_handle_by_sn.emplace(
+        dev_info.serial,
+        dds_device_handle{ dev_info,
+                           _publisher->create_datawriter( _topic, wqos, writer_listener.get() ),
```

Comment:
_publisher might be null if add_device() called before run()

Reply:
Hmm I changed the behavior of the ctor, run, add_device and remove_device to throw if called wrongly.